src/main/java/com/amazon/ion/impl/IonReaderContinuableCoreBinary.java

```diff
@@ -1233,7 +1233,9 @@ private void installMacros() {
          * Install any new symbols and macros, step out of the encoding directive, and resume reading raw values.
          */
         private void finishEncodingDirective() {
-            resetSymbolTable(); // TODO handle appended symbols
+            if (!isSymbolTableAppend) {
```

Comment:
This is a case where writing the code to do what the TODO said to do would have been faster than writing the TODO itself.